CHANGELOG.md

```diff
@@ -4,6 +4,30 @@
 
 ## DFX
 
+### feat: Integration of GitHub Action for `dfx` installation
+
+A new functionality has been implemented to enable the installation of `dfx` through GitHub Actions. This provides a more streamlined approach to installation:
+
+```yml
+jobs:
+  example-job:
+    runs-on: ubuntu-latest  # `macOS` works too by specifying `macos-latest`
+    steps:
+    - name: Install dfx
+      uses: dfinity/sdk@master
+    - name: Confirm successful installation
+      run: dfx --version
+```
+
+Moreover, it accommodates the installation of specific dfx versions:
+
+```yml
+    - name: Install dfx
+      uses: dfinity/sdk@master
+      with:
+        dfx-version: "0.14.2-beta.2"
```

Comment:
I see where you're coming from here:

- We're using the github action defined on the master branch
```
      uses: dfinity/sdk@master
```

- to install dfx version 0.14.2-beta.2
```
      with:
        dfx-version: "0.14.2-beta.2"
```

However, it seems like it would be natural to read the following as "install dfx as built from the sdk master branch"
```
    - name: Install dfx
      uses: dfinity/sdk@master
```

Do we have an option to do the following?

```
    - name: Install dfx at currently promoted version
      uses: dfinity/sdk
```


```
    - name: Install dfx at version 0.14.2-beta.2
      uses: dfinity/sdk@0.14.2-beta.2
```

Reply:
Sadly `uses: dfinity/sdk` won’t work, I was hoping for it too - GHActions require to specify a git ref. We could do `uses: dfinity/sdk@install-dfx` for example (install-dfx would need to be a branch or a tag). Or `@latest`

Reply:
Wouldn't that be possible if we officially [publish the action on the marketplace](https://docs.github.com/en/actions/creating-actions/publishing-actions-in-github-marketplace)? Similar to `actions/checkout@v3.5.3` we might be able to pin the version to `dfinity/sdk@v0.14.1`.

Reply:
It seems that `@ref` is mandatory, so we are always going to have one version for the `action` and one version for the actual `dfx` to install. This might be confusing.

We could have a kind of fixed ref for the action (e.g. `sdk@install-dfx` as suggested by Marcin) that we will update/overwrite if the action needs to be modified. Then, we can specify the version to install by using the `with` statement.

btw, `actions/checkout@v3` is very similar to this action. It has a certain version (`v3`) and it can refer other versions with the `ref` parameter.

Reply:
If we're already discussing the naming convention, what if we create a separate repository specifically for hosting the action, like `dfinity/setup-dfx`? This naming convention is used by other actions that set up environments, such as [`actions/setup-node`](https://github.com/marketplace/actions/setup-node-js-environment), [`actions/setup-python`](https://github.com/marketplace/actions/setup-python) and [`hashicorp/setup-terraform`](https://github.com/marketplace/actions/hashicorp-setup-terraform). Using a dedicated repository aligns with the official approach for publishing actions on the marketplace and doesn't require additional review from GitHub. While it won't solve the versioning problem, it shouldn't be confusing since this syntax is commonly used by other actions.

Reply:
Great insight.  Let's do that.

Reply:
I've requested to change the visibility of https://github.com/dfinity/setup-dfx to `public` and I'll notify once it's public

Reply:
@mmicu https://github.com/dfinity/setup-dfx is now public 🎉